src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/CallSiteRuntimeResolver.cs

```diff
@@ -92,7 +89,7 @@ protected override object VisitScopeCache(ServiceCallSite callSite, RuntimeResol
         private object VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
         {
             bool lockTaken = false;
-            Dictionary<ServiceCacheKey, object> resolvedServices = serviceProviderEngine.ResolvedServices;
+            ConcurrentDictionary<ServiceCacheKey, object> resolvedServices = serviceProviderEngine.ResolvedServices;
```

Comment:
This should be `IDictionary<ServiceCacheKey, object>`.

Reply:
OK